gptme/llm/llm_anthropic.py

```diff
@@ -358,22 +357,35 @@ def _transform_system_messages(
     # unless a `call_id` is present, indicating the tool_format is 'tool'.
     # Tool responses are handled separately by _handle_tool.
     for i, message in enumerate(messages):
-        if message.role == "system" and message.call_id is None:
+        if message.role == "system":
+            content = (
+                f"<system>{message.content}</system>"
+                if message.call_id is None
+                else message.content
+            )
+
             messages[i] = Message(
                 "user",
-                content=f"<system>{message.content}</system>",
+                content=content,
                 files=message.files,  # type: ignore
+                call_id=message.call_id,
             )
 
     # find consecutive user role messages and merge them together
     messages_new: list[Message] = []
     while messages:
         message = messages.pop(0)
-        if messages_new and messages_new[-1].role == "user" and message.role == "user":
+        if (
+            messages_new
+            and messages_new[-1].role == "user"
+            and message.role == "user"
+            and message.call_id == messages_new[-1].call_id
```

Comment:
Why is this condition needed?

Reply:
When using the function call API, multiple calls may occur within a single answer. To handle this properly, ensure there is one tool response per tool call. By verifying that the call_id is distinct, we can ensure messages are not merged unless they originate from the same function call.